File: Cap 05/5.5 - Laco Condicional Seletivo/Simulado/Com Enquanto-infinito/enqinfit.dart
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.5
 * 
 * Programa ..: LACO SELETIVO ITERATIVO (SIMULADO ENQUANTO)
 * 
 */
 
import 'dart:io';

int I = 0, N = 0, R = 0;
void main() {
  print("LACO PRE-TESTE COM FLUXO VERDADEIRO");
  print("Multiplicacao de numero por 3 por 5 vezes\n");
  I = 1;
  while (true) { // simulacao "laco"
    print("Calculo: $I");
    stdout.write("Entre valor numerico inteiro: ");
    N = int.parse(stdin.readLineSync()!);
    R = N * 3;
    print("O numero informado X 3 = $R\n");
    if (I > 4) {break;} // simulacao "saia_caso (I > 4)"
    I = I + 1;
  } // simulacao "fim_laco"
}

File: Cap 05/5.5 - Laco Condicional Seletivo/Simulado/Com Enquanto-infinito/enqinfit.php
<?php
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 5 - Topico 5.5
 * 
 * Programa ..: LACO SELETIVO ITERATIVO (SIMULADO ENQUANTO)
 * 
 */

$I;
$N;
$R;
echo "LACO PRE-TESTE COM FLUXO VERDADEIRO" . PHP_EOL;
echo "Multiplicacao de numero por 3 por 5 vezes" . PHP_EOL . PHP_EOL;
$I = 1;
while (true) {
    echo "Calculo: " . $I . PHP_EOL;
    echo "Entre valor numerico inteiro: ";
    $N = (int) readline();
    $R = $N * 3;
    echo "O numero informado X 3 = " . $R . PHP_EOL . PHP_EOL;
    if ($I > 4) break;
    $I = $I + 1;
}
return 0;

?>
